src/pages/NewsFeed.js

```diff
@@ -1,5 +1,45 @@
+import { useState, useEffect } from "react";
+import PostList from "../components/PostList";
+
 function NewsFeed() {
-  return <h1>News Feed</h1>;
+  const [isLoading, setIsLoading] = useState(true);
+  const [loadedPosts, setLoadedPosts] = useState([]);
+
+  useEffect(() => {
+    fetch(
+      "https://friendface-react-default-rtdb.europe-west1.firebasedatabase.app/posts.json"
+    )
+      .then((response) => {
+        return response.json();
+      })
+      .then((data) => {
+          const posts = [];
+          for (const key in data) {
+              const post = {
+                  id: key,
+                  ...data[key]
+              }
+              posts.push(post);
+          }
+        setIsLoading(false);
+        setLoadedPosts(posts);
+      });
+  }, []);
```

Comment:
I'm concerned that there is nothing in the dependency array. This useEffect will update a state (from setIsLoading/LoadedPosts) then that will trigger this useEffect, and then it will call every time. Is there anything you could use so that it only runs when it is worth doing so?

Reply:
The empty array is there so that the function only runs a first time when the component runs, afterwards it won't run again (and I don't need it to at this point) as the array will be equal on each run for the useEffect.

Reply:
Nice, it looks counterintuitive but it does ring a bell on how it works. Great display of knowledge here!